dace/sdfg/replace.py

```diff
@@ -24,6 +25,20 @@ def _replsym(symlist, symrepl):
                           if symbolic.issymbolic(dim) else dim)
     return symlist
 
+def deepreplace(subgraph: 'dace.sdfg.state.StateGraphView', name: str,
+                new_name: str):
```

Comment:
I'm a bit uncomfortable with this: The reason why we don't normally traverse into nested SDFGs when doing replacement is that symbols don't necessarily mean the same thing, `i` could refer to one thing in one SDFG, and something else in another.

Why did you need this? Perhaps we can find another solution, using `sdfg.constants` and `nested_sdfg.symbol_mapping` to solve this without doing the replacement in nested SDFGs? Or, at least, we should make sure that the symbol _does_ mean the same thing (by checking that it is not defined again in the nested SDFG and that the symbol mapping doesn't map it to something else).

Reply:
Assuming I would just add the value to sdfg.constants before generating it and afterwards removing them again, would this behave as expected? (I.e. use the constant if it's not defined, otherwise take whatever it is in the nested sdfgs)

Reply:
It _should_ work with nested SDFGs, I'm just saying that we need to be more careful there because of the name clash I mentioned above. If you fix the variable `w` in the outer SDFG, and then do a symbol mapping of, e.g., `{"w": "w"}` in the nested SDFG, does this still not produce a constant?

If not, then I would prefer the solution of adding them to `sdfg.constants` as you suggested.

Reply:
I already implemented it now with constants. Should be fine. Copied from http://candcplusplus.com/c-variables-scopelocal-variables-and-global-variables:
"If a global and local variable have the same name and if we try to access the variable then note the local scope variable with the same name has a greater influence in that local region and so it will be used.However,if the local variable cease to exist then the global variable will be used instead.Consider the example below." Since dace always creates local regions whenever a variable is created, it should work out, but I will create a little addition to the test to verify it

Reply:
So now we got rid of `deepreplace`? Did we even get rid of the backup stuff as well?